go/batch-submitter/drivers/sequencer/encoding_test.go

```diff
@@ -292,6 +151,29 @@ func testAppendSequencerBatchParamsEncodeDecode(
 	paramsBytes, err := params.Serialize()
 	require.Nil(t, err)
 	require.Equal(t, test.HexEncoding, hex.EncodeToString(paramsBytes))
+
+	// Serialize the batches in compressed form
+	params.Type = sequencer.BatchTypeZlib
+	compressedParamsBytes, err := params.Serialize()
+	require.Nil(t, err)
+
+	// Deserialize the compressed batch
+	var paramsCompressed sequencer.AppendSequencerBatchParams
+	err = paramsCompressed.Read(bytes.NewReader(compressedParamsBytes))
+	require.Nil(t, err)
+	require.Equal(t, paramsCompressed.Type, sequencer.BatchTypeZlib)
+
+	// Handle nil contexts, Read() will never return nil contexts
+	if expParams.Contexts == nil {
```

Comment:
I think if `Contexts == nil`, it should return an `errMalformedBatch`. I also think that if `len(Contexts) == 0 && len(Transactions) > 0` it should return the error. Same with if the total number of elements to append is not equal to the actual number of elements in the batch.

Thoughts @cfromknecht?

Reply:
I've added a malformed batch error that checks to make sure the batch is shaped in a way that is meaningful to the protocol